### PR TITLE
Bluetooth: PACS: Add PAC data to pac_notify

### LINKED ISSUE #41191: LE Audio: Update pac_indicate to actually send data
pac_indicate has not been implemented to send any actual data, where it should send the PAC records.

### LINKED ISSUE #41192: LE Audio: Change PACS from indicate to notify
PACS is implemented as indicating changes, rather than notifying, making it incompatible with the PACS spec.

Change from bt_gatt_indicate() to bt_gatt_notify()

### PR DESCRIPTION
**Pull request by Thalley**
This fixes the pac_notify (formerly pac_indicate) to actually notify the PAC records to the connected clients. 

fixes https://github.com/zephyrproject-rtos/zephyr/issues/41191
fixes https://github.com/zephyrproject-rtos/zephyr/issues/41192

### CODE REVIEW

File: include/bluetooth/audio/audio.h [hunk 1]
```diff
@@ -1057,6 +1057,11 @@ struct bt_audio_unicast_server_cb {
 	 *  is returned, the previously returned @p codec values (if any) will
 	 *  be sent to the client that requested the value.
 	 *
+	 *  @param conn   The connection that requests the capabilities.
+	 *                Will be NULL if the capabilities is requested for
+	 *                sending a notification, as a result of callling
+	 *                bt_audio_capability_register() or
+	 *                bt_audio_capability_unregister().
```
Comment (MariuszSkamra) [inline include/bluetooth/audio/audio.h:1064]: Looks unrelated, thus I'd suggest to push this change it in another commit

**Review by asbjornsabo — DISMISSED** [review]

**Review by jhedberg — DISMISSED** [review]

**Review by MariuszSkamra — DISMISSED** [review]

**Review by asbjornsabo — DISMISSED** [review]

**Review by asbjornsabo — DISMISSED** [review]

**Review by Casper-Bonde-Bose — APPROVED** [review]
lgtm

**Comment by Thalley** [comment]
@jhedberg @MariuszSkamra please re-review

**Review by Casper-Bonde-Bose — DISMISSED** [review]
LGTM

**Review by LingaoM — APPROVED** [review]

**Review by Casper-Bonde-Bose — APPROVED** [review]
LGTM

**Review by jhedberg — APPROVED** [review]

**Review by MariuszSkamra — APPROVED** [review]
